src/common/exif.cc

```diff
@@ -1378,6 +1378,7 @@ int dt_exif_read_blob(uint8_t *buf, const char *path, const int imgid, const int
       memcpy(buf + 6, &(blob[0]), length);
       return length + 6;
     }
+    fprintf(stderr, "[exif] Warning: too much data to store (%d bytes), Exif data will be discarded\n", length);
```

Comment:
My guess would be that `dt_exif_read_blob()` should have first parameter `uint8_t **blob`
Allocate enough memory with malloc(), and set *blob to the allocated memory.